frontend/packages/console-app/src/plugin.tsx

```diff
@@ -254,6 +258,28 @@ const plugin: Plugin<ConsumedExtensions> = [
         ).default,
     },
   },
+  // TODO: only show if one or more ConsolePlugins in installed
+  {
+    type: 'Dashboards/Overview/Health/Resource',
+    properties: {
+      // t('console-app~Console Plugins')
+      title: '%console-app~Console Plugins%',
+      resources: {
+        consolePlugins: {
+          kind: referenceForModel(ConsolePluginModel),
+          namespaced: false,
+          isList: true,
+        },
+      },
+      healthHandler: getDynamicPluginHealthState,
+      // t('console-app~Plugin status')
+      popupTitle: '%console-app~Plugin status%',
+      popupComponent: () =>
+        import(
```

Comment:
Type checking is resulting in the following error:
```ERROR in /Users/rhamilto/Git/console/frontend/packages/console-app/src/plugin.tsx(278,9):
TS2322: Type 'Promise<ComponentClass<Partial<WatchK8sResults<ResourcesObject>>, any> | FunctionComponent<Partial<WatchK8sResults<ResourcesObject>>> | FC<...>>' is not assignable to type 'Promise<ComponentType<Partial<WatchK8sResults<ResourcesObject>>>>'.
  Type 'ComponentClass<Partial<WatchK8sResults<ResourcesObject>>, any> | FunctionComponent<Partial<WatchK8sResults<ResourcesObject>>> | FC<...>' is not assignable to type 'ComponentType<Partial<WatchK8sResults<ResourcesObject>>>'.
    Type 'FC<WatchK8sResults<WatchConsolePluginsResource>>' is not assignable to type 'ComponentType<Partial<WatchK8sResults<ResourcesObject>>>'.
      Type 'FunctionComponent<WatchK8sResults<WatchConsolePluginsResource>>' is not assignable to type 'FunctionComponent<Partial<WatchK8sResults<ResourcesObject>>>'.
        Types of parameters 'props' and 'props' are incompatible.
          Type 'PropsWithChildren<Partial<WatchK8sResults<ResourcesObject>>>' is not assignable to type 'PropsWithChildren<WatchK8sResults<WatchConsolePluginsResource>>'.
            Property 'consolePlugins' is missing in type 'PropsWithChildren<Partial<WatchK8sResults<ResourcesObject>>>' but required in type 'WatchK8sResults<WatchConsolePluginsResource>'.
Version: typescript 3.8.3
Time: 48591 ms
```
@andrewballantyne, can you help me decrypt?  VS Code doesn't flag this, nor does the commit hook.  Am very confused.

Reply:
@rhamilto If you use a Dynamic Plugin Extension instead of a Static Plugin Extension... does this issue go away?

The error is saying someone (either the Static Extension or your `ConsolePluginPopover`) is using `Partial` in their type and removing `consolePlugins` but the other needs it part of their type. I may not be needed at runtime, but the types believe it is required.

Reply:
I did a quick investigation. You're not using the same type in your component. You cannot have a typed component as B and expect to give it to A. The code will try to give "A Props" when you want "B Props" ... which is going to get you an unexpected set of props and thus make the TS type for your component useless.

Reply:
Ah okay, so here is the thing. The type under the hood that you're comparing it to

```
// This is the Static Plugin type for popupComponent
popupComponent?: LazyLoader<WatchK8sResults<R>>;
. . .
// This is the definition of LazyLoader
type LazyLoader<T extends {} = {}> = () => Promise<React.ComponentType<Partial<T>>>;
```
Now note the `Partial<T>`. This is likely done originally to help soften the blow of unknown typings in play during the Static Plugin types. That way, in most cases, you won't have an issue. However, in order to do this, YOUR type:
```
type WatchConsolePluginsResource = {
  consolePlugins: ConsolePluginKind[];
};
```
Needs to have an equal ambiguity of how much it is expecting. The `Partial` type is like saying "everything is optional, despite what each property actually says".

So to fix this, you need to make your property `optional` as the Static Plugin typings won't allow you to operate on that.

I'm unsure if you'll run into this in the Dynamic Plugins, but it appears not as it's not a partial representation of what was given:
```
popupComponent?: CodeRef<WatchK8sResults<T>>;
. . .
type CodeRef<T = any> = () => Promise<T>;
```

Reply:
Thanks!  Making `consolePlugins` optional fixes it for both static and dynamic!